src/test/java/com/fakng/fakngagrgtr/parser/AmazonParserTest.java

```diff
@@ -0,0 +1,106 @@
+import reactor.core.publisher.Mono;
+
+@ExtendWith(MockitoExtension.class)
+public class AmazonParserTest extends AbstractParserTest {
+
+    private static final String URL = "Amazon";
+    private AmazonParser amazonParser;
+    @Mock
+    private CompanyRepository companyRepository;
+    @Mock
+    private LocationProcessor locationProcessor;
+
+    @BeforeEach
+    public void init() throws IOException {
+        String json = IOUtils.toString(Objects.requireNonNull(this.getClass()
+                .getResourceAsStream("amazon-data.json")), StandardCharsets.UTF_8);
+        WebClient webClient = WebClient.builder()
+                .exchangeFunction(clientRequest -> Mono.just(
+                        ClientResponse.create(HttpStatus.OK)
+                                .header("content-type", "application/json")
+                                .body(json)
+                                .build()
+                )).build();
+        amazonParser = new AmazonParser(webClient, companyRepository, locationProcessor, URL);
+    }
+
+    @Test
+    public void testParseCreatesVacanciesFromJsonWithCachedLocations() {
+        Company company = prepareCompany();
+        Mockito.when(companyRepository.findByTitle(URL))
+                .thenReturn(Optional.of(company));
+        Mockito.when(locationProcessor.processLocation(company, "city_1", "country_1"))
+                .thenReturn(company.getLocations().get(0));
+        Mockito.when(locationProcessor.processLocation(company, "city_2", "country_2"))
+                .thenReturn(company.getLocations().get(1));
+        amazonParser.init();
+        List<Vacancy> expectedVacancies = prepareVacancies(company);
+
+        List<Vacancy> actualVacancies = amazonParser.parse();
+
+        assertEquals(expectedVacancies.size(), actualVacancies.size());
+        for (int i = 0; i < expectedVacancies.size(); i++) {
+            assertVacancy(expectedVacancies.get(i), actualVacancies.get(i));
+        }
+    }
+
+    private List<Vacancy> prepareVacancies(Company company) {
+        List<Vacancy> vacancies = new ArrayList<>();
+
+        Vacancy first = new Vacancy();
+        first.setTitle("title_1");
+        first.setUrl("https://www.amazon.jobs" + "/url_1");
+        first.setCompany(company);
+        first.setDescription("description 1\nbasic_qualifications 1\npreferred_qualifications 1");
+        first.addLocation(company.getLocations().get(0));
+        vacancies.add(first);
+
+        Vacancy second = new Vacancy();
+        second.setTitle("title_2");
+        second.setUrl("https://www.amazon.jobs" + "/url_2");
+        second.setCompany(company);
+        second.setDescription("description 2\nbasic_qualifications 2\npreferred_qualifications 2");
+        second.addLocation(company.getLocations().get(1));
+        vacancies.add(second);
+
+        return vacancies;
+    }
+
+    private void assertVacancy(Vacancy expected, Vacancy actual) {
```

Comment:
I feel that if we reuse methods and code from GoogleParserTest, we should consider moving duplicated parts/methods in a base class for these test classes and re-use those :)

Reply:
Done.